deepchecks/vision/checks/performance/single_dataset_scalar_performance.py

```diff
@@ -0,0 +1,148 @@
+        """Initialize the metric for the check, and validate task type is relevant."""
+        if self.metric is None:
+            if context.train.task_type == TaskType.CLASSIFICATION:
+                self.metric = Accuracy()
+                if not hasattr(self, 'metric_name'):
+                    self.metric_name = 'accuracy'
+            elif context.train.task_type == TaskType.OBJECT_DETECTION:
+                self.metric = ObjectDetectionAveragePrecision()
+                if not hasattr(self, 'metric_name'):
+                    self.metric_name = 'object_detection_average_precision'
+                if self.reduce is None:
+                    self.reduce = torch.nanmean
+            else:
+                raise DeepchecksValueError('For task types other then classification or object detection, '
+                                           'pass a metric explicitly')
+        self.metric.reset()
+
+    def update(self, context: Context, batch: Batch, dataset_kind: DatasetKind.TRAIN):
+        """Update the metrics by passing the batch to ignite metric update method."""
+        label = batch.labels
+        prediction = batch.predictions
+        self.metric.update((prediction, label))
+
+    def compute(self, context: Context, dataset_kind: DatasetKind.TRAIN) -> CheckResult:
+        """Compute the metric result using the ignite metrics compute method and reduce to a scalar."""
+        metric_result = self.metric.compute()
+        if self.reduce is not None:
+            if isinstance(metric_result, float):
+                warnings.warn(SyntaxWarning('Metric result is already scalar, skipping reduce operation'))
+                result_value = metric_result
+            else:
+                result_value = float(self.reduce(metric_result))
+        elif isinstance(metric_result, float):
+            result_value = metric_result
+        else:
+            raise DeepchecksValueError(f'The metric {self.metric.__class__} return a non-scalar value, '
+                                       f'please specify a reduce function or choose a different metric')
+
+        metric_name = self.metric_name if hasattr(self, 'metric_name') else self.metric.__class__.__name__
+        if hasattr(self, 'reduce_name'):
+            reduce_name = self.reduce_name
+        elif self.reduce is None:
+            reduce_name = None
+        else:
+            reduce_name = self.reduce.__name__
+
+        result_dict = {'score': result_value,
+                       'metric': metric_name,
+                       'reduce': reduce_name}
+        return CheckResult(result_dict)
+
+    def add_condition_greater_than(self, threshold: float) -> ConditionResult:
+        """Add condition - the result is greater than the threshold."""
+        def condition(check_result):
+            if check_result['score'] > threshold:
+                return ConditionResult(ConditionCategory.PASS)
```

Comment:
We want to start passing details also on passing conditions. so maybe just do:
```
details = "Score found: {score}"
```
and pass it both pass & fail

Reply:
same for the rest of conditions

Reply:
+ 1 for that, also no need for if/else if you're returning a value